docs/design/TEMPLATE.md

```diff
@@ -0,0 +1,56 @@
+[This is a template for Hyperspace's design proposal process, documented [here](../../CONTRIBUTING.md).]
+
+# Proposal: [Title]
+
+Discussion at https://github.com/microsoft/hyperspace/XXXXX.
+
+## Abstract
+
+[A short summary of the proposal.]
+
+## Background
+
+[An introduction of the necessary background and the problem being solved by the proposed change.]
+
+## Proposal
+
+[A precise statement of the proposed change.]
+
+## Rationale
+
+[A discussion of alternate approaches and the trade offs, advantages, and disadvantages of the specified approach.]
+
+## Compatibility
+
+[A discussion of the change with regard to the
+[compatibility guidelines](../../COMPATIBILITY.md).]
+
+## Design
+
+[A description of the proposed design/algorithm. This should include a discussion of how the work fits 
+into [Hyperspace's roadmap](../ROADMAP.md).]
+
+> Note: If you want to use any images, please upload the .svg AND .png/.jpg file them to `/docs/design/img/` and link to them here.
+
+Here's a sample image:
+
+<img src="./img/sample.png"
+     alt="Sample image"
+     style="float: center; margin-right: 10px;"
+     width= "100px" />
+
+## Implementation
+
+[A description of the steps in the implementation, who will do them, and when.]
+
+> Note: If you want to use any images, please upload the .svg AND .png/.jpg file them to `/docs/design/img/` and link to them here.
+
+## Open issues (if applicable)
```

Comment:
Should we add a specific section/note to emphasize that if the feature could have perf impacts, we expect to see a reasonable plan to do perf evaluation to make sure no perf degrade occurs?

Reply:
Good idea. I've added it.